src/rust/engine/src/nodes.rs

```diff
@@ -1118,7 +1118,12 @@ impl Task {
                 ))
               }
             })?;
-          select.run_node(context).await
+          let res = select.run_node(context).await;
+          if get.safe && let Err(Failure::Throw { val, .. }) = res {
+            val
+          } else {
+            res
+          }
```

Comment:
@stuhood Hi!

I think you can imagine what I'm trying to do here, but my rust is rusty.. :) 

I've not been able to make rust to cooperate with me on this one.. guess I'll have to read some more rust docs 😆  but if you're able to say, oh that's easy just do X, than that would be very much appreciated :D 

Just to be clear, if `get.safe` is true, I want to translate any thrown exceptions as a regular result value (to be handled by rule code), anything else should just stay as normal.

Reply:
The types in each arm must be identical. You return a value plucked out of the error in the if and the whole result in the else.

Reply:
Wait, you have an enum already! Result. `Ok(val)` should do it.

Reply:
If you want to golf:
```rust
if let Err(Failure::Throw { val, .. }) = err {
  Ok(val)
} else {
  err
}
```

Reply:
Basically `if let ... this is a match destructure position ... = thing {`.

Reply:
Ah right, your example maybe started there - you can have compound conditionals yet where portions are let bindings unfortunately (yet? IIRC there is an RFC for this). You can add conditions to match arms though:
```rust
match res {
  Err(Failure::Throw { val, .. }) if get.safe => Ok(val),
  _ => res
}
```

Reply:
Aha: https://rust-lang.github.io/rfcs/2497-if-let-chains.html